### DISCUSSION TITLE
Limit metrics point amount

### DISCUSSION THREAD

**Discussion by xiangtianyu** [open-telemetry/opentelemetry-java#6801] [Q&A]
I used `-Dotel.experimental.metrics.cardinality.limit` to limit metrics cardinality, but i found histogram is different with other metrics type, if i set limit 5, histogram may upload about 5 * 15 = 75 records, is there some solution to set limit to the histogram and not effect the other metrics type

**Comment by trask** [comment]
You could use a metric view to override the buckets boundaries, in order to have less buckets

**Reply by jack-berg** [reply]
Yeah from the cardinality limit is indeed applied consistently to histograms as other instruments. The cardinality limits the total number of distinct metric points for a particular instrument. However, as you point out, histograms are different from other aggregations in that they contain more fields used to describe the distribution of measurement values, including: sum, count, min, max, and 0-N buckets where each bucket represents the count of measurements recorded with its boundaries. Bucket boundaries are the main thing that people configure with histograms, and there are a variety of tools for doing so:

- As @trask points out, you can use a [view](https://opentelemetry.io/docs/languages/java/sdk/#views)
- If you control the instrumentation, you can provide [advice](https://github.com/open-telemetry/opentelemetry-java/blob/537f5c4e79e99f82dae53d16f7167caf7041a0c3/api/all/src/main/java/io/opentelemetry/api/metrics/DoubleHistogramBuilder.java#L37-L47) to influence the default bucket boundaries used by the SDK
- You can override your metric reader's [default aggregation selector](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/MetricReader.java#L25) to change the default aggregation (including bucket boundaries) for all histogram instruments

**Reply by xiangtianyu** [reply]
Actually, i do not want to change the buckets boundaries, what i need is a dedicated configuration to limit the histogram count and not effect the other metrics types

**Reply by jack-berg** [reply]
What's the use case for limiting the the histogram count? Unless I'm misunderstanding, i don't think there's a way to achieve that.

**Reply by xiangtianyu** [reply]
Since histogram will generate much more records, it will cause more pressure on the server. 
When we set `-Dotel.experimental.metrics.cardinality.limit`, if we set limit for histogram, it's too restrictive on other type; if we set limit for other, there will be huge data, and will cause high load,

**Reply by trask** [reply]
I think you will be able to set a different cardinality limit for histograms using a metric view starting from the next release

check out
* https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/incubator#view-file-configuration
* https://github.com/open-telemetry/opentelemetry-java/pull/6794
